compiler/match_/inference/type_seeder.rs

```diff
@@ -114,12 +114,9 @@ impl<'this, Snapshot: ReadableSnapshot> TypeSeeder<'this, Snapshot> {
         context: &BlockContext,
         parent_vertices: &VertexAnnotations,
     ) -> Result<(), TypeInferenceError> {
-        self.local_variables(context, tig.conjunction.scope_id()).for_each(|var| {
-            let vertex = Vertex::Variable(var);
-            if let Some(parent_annotations) = parent_vertices.get(&vertex) {
-                tig.vertices.insert(vertex, parent_annotations.clone());
-            }
-        });
+        for (vertex, parent_annotations) in parent_vertices.iter() {
+            tig.vertices.insert(vertex.clone(), parent_annotations.clone());
+        }
```

Comment:
`local_variables` only considers variables declared at this depth (i.e. not referenced in any scope above). We do, however, still need to be able to access the type annotations of the captured variables.